src/router.rs

```diff
@@ -60,13 +51,10 @@ impl<T: Send + TryClone> HttpRoutes<T> {
     }
 
     /// Handle an incoming http request and generate corresponding response.
-    pub fn handle_http_request(&self, request: &Request, argument: &T) -> Response {
+    pub fn handle_http_request(&self, request: &Request, argument: T) -> Response {
```

Comment:
TryClone is indeed not needed. I don't understand though why you don't use a reference to argument? You would need to have a T that implements copy or move the `argument` ownership to the handle_http_request function.

Reply:
It's derived from the CloudHypervisor implementation. Yes, &T is more generic and will fix it.